static_inverse_kinematics.py

```diff
@@ -236,19 +226,20 @@ def animate(self):
         b.exec()
 
     def get_sol(self):
-        for i, sol in enumerate(self.sol):
-            for j, value in enumerate(sol.fun):
-                self.residuals_xyz[i][j] = value
-            self.nb_iteration_diff.append(sol.nfev)
-            self.nb_iteration_jac.append(sol.njev)
-
-        for j, value in enumerate(self.residuals_xyz):
-            for h in range(0, 3 * self.nb_markers, 3):
-                self.residuals[j][int(h / 3)] = np.linalg.norm(value[h : h + 3])
+        residuals_xyz = np.array([[0.0] * self.nb_markers * self.nb_dim] * self.nb_frames, ndmin=2)
+        residuals = np.array([[0.0] * self.nb_markers] * self.nb_frames, ndmin=2)
+        nb_iteration_diff = []
+        nb_iteration_jac = []
+
+        for i in range(self.nb_frames):
+            nb_iteration_diff.append(self.list_sol[i].nfev)
+            nb_iteration_jac.append(self.list_sol[i].njev)
+            residuals_xyz[i] = self.list_sol[i].fun
+            for h in range(0, self.nb_dim * self.nb_markers, self.nb_dim):
+                residuals[i][int(h / self.nb_dim)] = np.linalg.norm(self.list_sol[i].fun[h: h + self.nb_dim])
```

Comment:
I'm sure this loop is not useful,
Plus, I dislike the form `residuals[i][int(h / self.nb_dim)]`, I would recommend `residuals[i, int(h / self.nb_dim)]`
maybe you should explain to me what you're doing.
I feel it is not the right thing.

Reply:
done